src/Writers/HTMLWriter.jl

```diff
@@ -1405,6 +1405,30 @@ function generate_version_file(versionfile::AbstractString, entries, symlinks =
     end
 end
 
+# write redirect file
+function generate_redirect_file(redirectfile::AbstractString, entries, devurl, versions)
+    # If `versions` contains "stable"=>"v^" and `entries` contains "stable", then version="stable".
+    # In other cases, version="dev".
+
+    comment = "<!--This file is automatically generated by Documenter.jl-->"
+
+    isfile(redirectfile) && !startswith(read(redirectfile, String), comment) && return
+
+    version = devurl
+    stable_index = findfirst(v -> last(v)=="v^", versions)
+    if !isnothing(stable_index)
+        stable_version = first(versions[stable_index])
+        if stable_version in entries
+            version = stable_version
+        end
+    end
```

Comment:
In the current script, only `"stable"=>"v^" in versions` are supported. Should we take care of `"v#"`, `"v#.#"`and `"v#.#.#"`?